source/code/source/playfab/PlayFabCurlHttpPlugin.cpp

```diff
@@ -157,7 +157,7 @@ namespace PlayFab
 
         // Send
         curl_easy_setopt(curlHandle, CURLOPT_SSL_VERIFYPEER, true);
-        const auto res = curl_easy_perform(curlHandle);
+        const auto& res = curl_easy_perform(curlHandle);
```

Comment:
>const auto& [](start = 8, length = 11)

curl_easy is a c-api I'm pretty sure which means that there's no circumstance where the value being returned is ever a reference type, so similar to my other comment this doesn't have much value and may be confusing to others.

I'm not here to impose bumblelion style on the playfab codebase, but something we do (which I've really taken a shine to) is to only allow use of auto when the type is clearly present on the right-hand side of the initialization expression

e.g.
`auto x = static_cast<int*>(someVoidPtr);`

but all other places have to use full types

`<the actual curl type is> res = curl_easy_perform(...)`

Reply:
from your auto rules in the description it sounds like you'd prefer using the full type here is as well since it's not obvious from the right side.

---
In reply to: [340921208](https://github.com/PlayFab/XPlatCoreTemplate/pull/33#discussion_r340921208) [](ancestors = 340921208)